osu.Game/Modes/LegacyReplay.cs

```diff
@@ -0,0 +1,274 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using osu.Framework.Input;
+using osu.Framework.MathUtils;
+using osu.Game.Input.Handlers;
+using osu.Game.IO.Legacy;
+using OpenTK;
+using OpenTK.Input;
+using KeyboardState = osu.Framework.Input.KeyboardState;
+using MouseState = osu.Framework.Input.MouseState;
+
+namespace osu.Game.Modes
+{
+    public class LegacyReplay : Replay
+    {
+        protected List<LegacyReplayFrame> Frames = new List<LegacyReplayFrame>();
+
+        protected LegacyReplay()
+        {
+
+        }
+
+        public LegacyReplay(StreamReader reader)
+        {
+            float lastTime = 0;
+
+            foreach (var l in reader.ReadToEnd().Split(','))
+            {
+                var split = l.Split('|');
+
+                if (split.Length < 4 || float.Parse(split[0]) < 0) continue;
+
+                lastTime += float.Parse(split[0]);
+
+                Frames.Add(new LegacyReplayFrame(
+                    lastTime,
+                    float.Parse(split[1]),
+                    384 - float.Parse(split[2]),
+                    (LegacyButtonState)int.Parse(split[3])
+                    ));
+            }
+        }
+
+        public override ReplayInputHandler GetInputHandler() => new LegacyReplayInputHandler(Frames);
+
+        /// <summary>
+        /// The ReplayHandler will take a replay and handle the propagation of updates to the input stack.
+        /// It handles logic of any frames which *must* be executed.
+        /// </summary>
+        public class LegacyReplayInputHandler : ReplayInputHandler
+        {
+            private readonly List<LegacyReplayFrame> replayContent;
+            int currentFrameIndex;
+
+            public LegacyReplayFrame CurrentFrame => !hasFrames ? null : replayContent[currentFrameIndex];
+            public LegacyReplayFrame NextFrame => !hasFrames ? null : replayContent[MathHelper.Clamp(currentDirection > 0 ? currentFrameIndex + 1 : currentFrameIndex - 1, 0, replayContent.Count - 1)];

```

Comment:
Maybe separate this into a `nextFrameIndex` property and this one. Makes the line easier to parse mentally.